### PR TITLE
complete Calculator project

### PR DESCRIPTION
**Pull request by govind201**
# Description
This project involves building a basic HTML calculator with functions for addition, subtraction, multiplication, and division. The calculator operation will consist of a number, an operator, and another number, and the operate function will take an operator and two numbers and call the appropriate function. The user interface will include buttons for each digit, the basic math functions, an "equals" key, and a "clear" button, as well as a display for the calculator's output.

# Screenshots
image

image

image

# Features
* Supports basic math functions: addition, subtraction, multiplication, and division.
* Allows users to string together multiple operations and evaluates each pair of numbers at a time.
*  Includes a "clear" button to wipe out existing data and start fresh.
* Handles edge cases such as rounding long decimals and preventing division by zero.

# Testing
* Test each of the basic math functions (add, subtract, multiply, divide) with different sets of input values to ensure they produce the expected results.
* Test the functionality of the "equals" key by entering various combinations of numbers and operators to verify that the calculator correctly evaluates the expression and displays the correct result.
* Test the "clear" button by entering some numbers and operations, clicking "clear", and verifying that the calculator display is empty and ready for new input.
* Test the handling of edge cases such as division by zero, long decimals, and invalid input (such as pressing "=" before entering any numbers or operators) to ensure that the calculator responds appropriately and doesn't crash.

### CODE REVIEW

**Comment by devanuj97** [comment]
@govind201 please try to take feedback seriously i'm reminding this second time to you!, from now please be mindful and try to add assignment link and PR live link in the PR description, 
As discussion before rasing PR test yourself first whether all functionality is working fine or not at minimum 

Here while doing 1:1 i found no valid answer why you have added[ this](https://github.com/govind201/calculator-project/pull/1/files#diff-e07d531ac040ce3f40e0ce632ac2a059d7cd60f20e61f78268ac3be015b3b28fR53) function -- based on you answer it seems like you define it for some reason but forgot to use this functionality 

leaving this PR review in the middle expecting you will test every node with your own first(as a part of self review)

File: app.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+const clear = () => {
+    firstNum = null;
+    secondNum = null;
+    operator = null;
+    result = null;
+    display.textContent = "0";
+}
+
+const handleNumberClick = (e) => {
```
Comment (devanuj97) [inline app.js:51]: We should avoid using if-else ladder as much as possible, this is not good for code readability and sometime cause error-prone 
as discuss try to refactor this

File: app.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            secondNum = number;
+        } else {
+            secondNum += number;
+        }
+        display.textContent = `${firstNum} ${operator} ${secondNum || ""}`;
+    }
+}
+
+const handleOperatorClick = (e) => {
+    operator = e.target.value;
```
Comment (devanuj97) [inline app.js:71]: can we refactor this as well and avid if-else ladder until this is the only option we have

File: app.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            if (secondNum === null) {
+                secondNum = "0.";
+            } else {
+                secondNum += ".";
+            }
+            display.textContent = secondNum;
+        }
+    }
+}
+numberBtns.forEach((btn) => {
```
Comment (devanuj97) [inline app.js:119]: add space above

File: app.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    btn.addEventListener("click", handleNumberClick);
+});
+operatorBtns.forEach((btn) => {
```
Comment (devanuj97) [inline app.js:122]: add space above

File: app.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    btn.addEventListener("click", handleOperatorClick);
+});
+equalsBtn.addEventListener("click", handleEqualsClick);
```
Comment (devanuj97) [inline app.js:125]: add space above

**Review by devanuj97 — CHANGES_REQUESTED** [review]

**Review by devanuj97 — APPROVED** [review]
